### ISSUE TITLE
"KeypadTones" setting missing from Retevis RT3S codeplugs

### ISSUE THREAD

**Issue by tylert** [DaleFarnsworth-DMR/codeplug#2]
I'm not certain if this is the correct repo in which to raise this issue, however, this is where the code change is likely to be made so I'll start here first @DaleFarnsworth.

When using Editcp to upload a new codeplug to a Retevis RT3S radio (SKU A9110F;  the GPS version running firmware P018.016), the "KeypadTones" setting is missing and will always be reset to "On".  The "DisableAllTones" and "TalkPermitTone" settings seem to work as expected.

I suspect, but am not 100% certain, that the fix is in this location of the code:
```
diff --git a/generated.go b/generated.go
index 22fd767..8a1239e 100644
--- a/generated.go
+++ b/generated.go
@@ -1295,6 +1295,7 @@ var riGeneralSettings_uv380 = recordInfo{
                &fiGsDisableAllTones,
                &fiGsChFreeIndicationTone,
                &fiGsTalkPermitTone,
+               &fiGsKeypadTones,
                &fiGsCallAlertToneDuration,
                &fiGsScanDigitalHangTime,
                &fiGsScanAnalogHangTime,
```

I'd love to help out but am just missing a few essential build/install instructions necessary for me to assist in debugging this code/feature (i.e.:  "make" fails due to missing "qtdeploy").  I'm currently working on a Debian 10.x amd64 machine.

**Comment by tylert** [comment]
Nearly forgot... I'm currently running editcp 1.0.18 downloaded from https://www.farnsworth.org/dale/codeplug/editcp/.

**Comment by DaleFarnsworth** [comment]
Thank you Tyler.  That is indeed where the update should.  Actually, that file is generated programmatically, hence the name.  But when I added the field to codeplugs.json and ran the program to create generated.go, it made the exact change in your diff above.

I made a v1.0.19 containing your change.  Please give it a try!

**Comment by tylert** [comment]
Fantastic!  I'll try the new version when I get "home" from work :)

Even though my previous attempt to build Editcp failed, I've been successful building dmrConfig from source on my local machine (you need to rename all the directories if you're downloading them from GitHub as archives or you need to check out clones of each component locally and make sure they're named appopriately).

I'm a big fan of Editcp (sharing it with 20+ new DMR owners) and I hope to raise/fix a few more new issues I found relating while working with the Retevis RT3S (e.g.: firmware uploading, contacts uploading, etc.).

Actually, I would also love to see some slightly more verbose README files in each of the repos used for Editcp outlining the build steps, prerequisite packages, etc.

I'm also curious about the workflow you use when you build everything which might help others to iterate on a new codeplug format (e.g.:  "the Anytone HTs" or Radioddity GD-77/Tytera MD-760).

I'd also like to point out that https://www.farnsworth.org/dale/codeplug/editcp/ still links to the old GitHub repo instead of the new one(s) too.

**Comment by DaleFarnsworth** [comment]
Yes, using therecipe/qt with go is quite a pain.  Adding support for a new
radio involves editing codeplug/codeplugs.json, adding the new (empty)
codeplug files codeplug/new/ (by extracting new.tar.bz2 to codeplug/new/),
and then running make in codeplug/.

If the new radio has entirely new fields, you need to add the graphics
widgets to the gui so they can be modified.  This involves editing a file
in editcp/, e.g. editcp/generalSettings.go for a new General Settings field.

I regret splitting out the subdirs from the old DaleFarnsworth/codeplug
repo.  It created extra work, without any real benefit that I can see.
I've even toyed with the idea of reviving that repo and bringing it
up-to-date with the changes in DaleFarnsworth-DMR.  Sigh.

Thanks for pointing the link to the old repo.  I fixed it.

Dale

**Comment by tylert** [comment]
Joining git repos while preserving history is easy (looks a bit weird in the history since one repo will have a dozen "birth commits").  Lemmie know if you want help with this.

**Comment by DaleFarnsworth** [comment]
I know it can be done.  The only challenge will be preserving
bisectability, which I probably wouldn't bother doing.  I used git daily
for a dozen years (and bitkeeper before that) as a Linux kernel
programmer.  But I've gotten a bit rusty since retiring.  Thanks for
offering.  If I decide to do it, and run into problems, I'll get in touch.

**Comment by tylert** [comment]
Thanks, @DaleFarnsworth, tested the fix with 1.0.19 and it works great on the RT3S units.  Closing ticket.
